frontend/src/run.js

```diff
@@ -341,10 +343,17 @@ export class Run extends React.Component {
 
   getRun() {
     fetch(Settings.serverUrl + '/run/' + this.state.id)
-      .then(response => response.json())
+      .then(response => {
+        this.setState({ "httpStatus": response.status });
+        if (!response.ok) {
+          throw new Error("Failed with HTTP code " + response.status);
+        }
+        return response.json();
+      })
```

Comment:
Same comment here about `httpStatus` and the `isNotFound`/`isRunValid` state variable.